From febffb4e6227aa946db476634ebe39b22d4ee804 Mon Sep 17 00:00:00 2001
From: Dan Voiculeasa <dan.voiculeasa@windriver.com>
Date: Tue, 10 May 2022 12:21:32 +0300
Subject: [PATCH 11/11] Debian: Hardcode net.naming-scheme in CONFIG_CMDLINE

Between CentOS systemd v219 and Debian systemd v247 some changes for
determining NIC naming were introduced. Something called Predictable
Names is used[1]. When booting Debian we will see interfaces using a
different predictable name, which requires to be handled at an upper
level.

Try to keep differences to a minimum, to minimize upgrades impact.
Hardcode kernel parameter here to keep the effort contained.

On Debian we use systemd 247.
Use scheme v238, the oldest known scheme[2] to systemd 247.

There will be future work to handle the new naming scheme.

[1]: https://wiki.debian.org/NetworkInterfaceNames
[2]: https://github.com/systemd/systemd/blob/v247/src/
shared/netif-naming-scheme.c#L8

Signed-off-by: Dan Voiculeasa <dan.voiculeasa@windriver.com>
---
 debian/config/amd64/none/config | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/debian/config/amd64/none/config b/debian/config/amd64/none/config
index bb72bbd94..0d66ceff0 100644
--- a/debian/config/amd64/none/config
+++ b/debian/config/amd64/none/config
@@ -463,7 +463,8 @@ CONFIG_HOTPLUG_CPU=y
 CONFIG_LEGACY_VSYSCALL_EMULATE=y
 # CONFIG_LEGACY_VSYSCALL_XONLY is not set
 # CONFIG_LEGACY_VSYSCALL_NONE is not set
-# CONFIG_CMDLINE_BOOL is not set
+CONFIG_CMDLINE_BOOL=y
+CONFIG_CMDLINE="net.naming-scheme=vSTX7_0"
 CONFIG_MODIFY_LDT_SYSCALL=y
 CONFIG_HAVE_LIVEPATCH=y
 CONFIG_LIVEPATCH=y
-- 
2.36.1

